quickstart: Update to using kustomize with the ccruntime sample payloads

fidencio opened the pull request:
During the `v0.3.0` development cycle we've changed to using kustomize with the ccruntime and ssh-demo sample payloads, so we need to reflect it here.

As the `v0.3.0` is not yet released, one can test the changed by using `4a41c6fd66921a2a56bbf7762d92cf37ddf568ed` as ref, as shown below:
```
kubectl apply -k github.com/confidential-containers/operator/config/samples/ccruntime/default?ref=4a41c6fd66921a2a56bbf7762d92cf37ddf568ed
```
Or, for s390x:
```
kubectl apply -k github.com/confidential-containers/operator/config/samples/ccruntime/s390x?ref=4a41c6fd66921a2a56bbf7762d92cf37ddf568ed
```

fidencio committed:
quickstart: Adapt to the operator using kustomize

Let's adapt the instructions to using kustomize for deploying the sample
ccruntime custom resource.

Signed-off-by: Fabiano Fidêncio <fabiano.fidencio@intel.com>

fidencio committed:
quickstart: Use v0.3.0 as example, instead of v0.2.0

As we're about to release v0.3.0, let's update the quickstart guide so
it's easier for folks to try it out using the correct latest release.

Signed-off-by: Fabiano Fidêncio <fabiano.fidencio@intel.com>

fidencio committed:
quickstart: Update description of the qemu-tdx runtime class

Now it's prepared to be used with Verdictd and EAA KBC.

Signed-off-by: Fabiano Fidêncio <fabiano.fidencio@intel.com>

fidencio commented:
/cc @BbolroC @stevenhorsman @mythi @fitzthum

stevenhorsman reviewed (approved):
LGTM. I tested using (roughly) the following shell commands on a new 20.04 Ubuntu VSI:
```
sudo apt-get -y update && sudo apt-get -y install ansible
git clone https://github.com/confidential-containers/operator.git
cd operator/tests/e2e/
ansible-playbook -i localhost, -c local --tags untagged ansible/main.yml
sudo -E PATH="$PATH" bash -c './cluster/up.sh'
export KUBECONFIG=/etc/kubernetes/admin.conf
kubectl label node "$(hostname)" node-role.kubernetes.io/worker=
kubectl apply -k github.com/confidential-containers/operator/config/release?ref=1def978fdf6b0cb2383d43970c3dc484df3cad54
 ```
After waiting a little while I got:
```
# kubectl get pods -n confidential-containers-system
NAME                                              READY   STATUS    RESTARTS   AGE
cc-operator-controller-manager-79797456f6-x5lkv   2/2     Running   0          20s
```
I then created the custom resource with:
```
# kubectl apply -k github.com/confidential-containers/operator/config/samples/ccruntime/default?ref=1def978fdf6b0cb2383d43970c3dc484df3cad54
```
After a bit longer this time it seemed to work fine:
```
# kubectl get pods -n confidential-containers-system
NAME                                              READY   STATUS    RESTARTS   AGE
cc-operator-controller-manager-79797456f6-x5lkv   2/2     Running   0          2m8s
cc-operator-daemon-install-x9vms                  1/1     Running   0          79s
cc-operator-pre-install-daemon-rv25l              1/1     Running   0          94s
root@stevenh-big-daddy-operator:~/go/src/github.com/confidential-containers/operator/tests/e2e# kubectl get runtimeclass

NAME       HANDLER    AGE
kata       kata       9s
kata-clh   kata-clh   4s
```

The only thing I'll note is that I initially Icreated a VSI with 2vCPU & 8GiB RAM, which matches the pre-req in our doc: `Ensure a minimum of 8GB RAM and 2 vCPU for the Kubernetes cluster node`, but when I deploy the operator,  I hit an error:
```
# kubectl describe pods -n confidential-containers-system
Name:           cc-operator-controller-manager-79797456f6-spmss
Namespace:      confidential-containers-system
...
Events:
  Type     Reason            Age    From               Message
  ----     ------            ----   ----               -------
  Warning  FailedScheduling  4m12s  default-scheduler  0/1 nodes are available: 1 Insufficient cpu. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod.
```
When I re-ran with 4 vCPU and 16GB RAM is worked fine, but I'm not sure if something has changed that means we need to increase the VM size we run it on compared to the doc? For full disclosure, I normally use a 2vCPU, 8GB internal system, but that's down at the moment, so I used IBM Cloud.

fidencio commented:
> The only thing I'll note is that I initially Icreated a VSI with 2vCPU & 8GiB RAM, which matches the pre-req in our doc: `Ensure a minimum of 8GB RAM and 2 vCPU for the Kubernetes cluster node`, but when I deploy the operator, I hit an error

Oh, I could reproduce the same error here.

stevenhorsman commented:
Quick update - I've just realised that my normal test VM are 4CPU, 8GB RAM, not 2, so that might make the difference.

fidencio commented:
> Quick update - I've just realised that my normal test VM are 4CPU, 8GB RAM, not 2, so that might make the difference.

Everything does work as expected using 4CPUs, and I'm also updating the docs accordingly, thanks for bringing this up!

fidencio committed:
quickstart: Update the required vCPU numbers to 4

While preparing the `v0.3.0` release, we've noticed that using a VM with
2 vCPUs would lead to:
```
Name:           cc-operator-controller-manager-79797456f6-spmss
Namespace:      confidential-containers-system
...
Events:
  Type     Reason            Age    From               Message
  ----     ------            ----   ----               -------
  Warning  FailedScheduling  4m12s  default-scheduler  0/1 nodes are available: 1 Insufficient cpu. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod.
```

And this is *NOT* something introduced between `v0.2.0` and `v0.3.0`, as
it also happen with the previous release.

For now, let's update the documentation accordingly and revisit this
after the release in case we need to really rely on deploying in nodes
with 2 vCPUs.

Signed-off-by: Fabiano Fidêncio <fabiano.fidencio@intel.com>

mythi reviewed (approved):
/lgtm

fitzthum commented on quickstart.md:
````diff
@@ -61,9 +61,9 @@ with the desired [release tag](https://github.com/confidential-containers/operat
+For example, to deploy the `v0.3.0` release run:
 ```
-kubectl apply -k github.com/confidential-containers/operator/config/release?ref=v0.2.0
+kubectl apply -k github.com/confidential-containers/operator/config/release?ref=v0.3.0
````
Btw what is expected behavior if the ref is not included here? Is  there a default version?

fidencio replied:
The latest commit would be used.

fitzthum reviewed (approved):
LGTM
